panels/user-accounts/cc-avatar-chooser.c

```diff
@@ -411,77 +411,133 @@ cheese_camera_device_monitor_new_cb (GObject *source,
 }
 #endif /* HAVE_CHEESE */
 
-static void
-setup_photo_popup (CcAvatarChooser *self)
+static GStrv
+get_settings_facesdirs (void)
 {
-        GFile *file, *dir;
-        GFileInfo *info;
-        GFileEnumerator *enumerator;
-        GFileType type;
-        const gchar *target;
-        const gchar * const * dirs;
-        guint i;
-        gboolean added_faces;
+        g_autoptr(GSettingsSchema) schema = NULL;
+        g_autoptr(GSettings) settings = NULL;
+        g_auto(GStrv) settings_dirs = NULL;
+        g_autoptr(GPtrArray) facesdirs = NULL;
+        GSettingsSchemaSource *source = g_settings_schema_source_get_default ();
+
+        facesdirs = g_ptr_array_new ();
+
+        if (source) {
+                schema = g_settings_schema_source_lookup (source,
+                                                          "org.gnome.desktop.interface",
+                                                          FALSE);
+        }
 
-        self->faces = g_list_store_new (G_TYPE_FILE);
-        gtk_flow_box_bind_model (GTK_FLOW_BOX (self->flowbox),
-                                 G_LIST_MODEL (self->faces),
-                                 create_face_widget,
-                                 self,
-                                 NULL);
+        if (schema) {
+                settings = g_settings_new_with_path ("org.gnome.desktop.interface",
+                                                     "/org/gnome/desktop/interface/");
+                settings_dirs = g_settings_get_strv (settings, "facesdirs");
+
+                if (settings_dirs != NULL) {
+                        int i;
+                        for (i = 0; settings_dirs[i] != NULL; i++) {
+                                char *path = settings_dirs[i];
+                                if (path != NULL && g_strcmp0 (path, "") != 0)
+                                        g_ptr_array_add (facesdirs, g_strdup (path));
+                        }
+                }
+        }
 
-        g_signal_connect (self->flowbox, "child-activated",
-                          G_CALLBACK (face_widget_activated), self);
+        return (GStrv) g_steal_pointer (&facesdirs->pdata);
```

Comment:
https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/553/diffs#note_770390

Reply:
I've updated the PR adding a NULL pointer to the array.